cloud_AWS/terraform/module/examples/multiple-vpc-regions/aws_multiregions.py

```diff
@@ -19,33 +19,55 @@
 
 TerraformVars = Dict[str, Any]  # variables passed in terraform plan/apply/destroy call
 TerraformAction = Callable[[Terraform, TerraformVars], bool]  # terraform plan/apply/destroy
+Region = Dict[str, Dict]
 
 
-def execute_action(action: TerraformAction, vpc_regions: dict) -> bool:
+def validate_input_data(vpc_regions: Dict[str, Region]) -> bool:
+    if not vpc_regions.get("plan_id"):
+        print_log("plan_id is required.")
+        return False
+    if not vpc_regions.get("external_id"):
+        print_log("external_id is required.")
+        return False
+    if not vpc_regions.get("regions"):
+        print_log("At least one region is required.")
+        return False
+    return True
+
+
+def execute_action(action: TerraformAction, vpc_regions: Dict[str, Region]) -> bool:
     """Execute an action for every region"""
 
+    if not validate_input_data(vpc_regions):
+        return True
+
     t_bucket = Terraform(working_dir="bucket/")
     bucket_region_name = []
     bucket_arn_list = []
-    err_occur = False
+    no_err = True
```

Comment:
negative variables are harder to reason about. why this change?

Reply:
So I've changed it to `execution_success = True`

Reply:
Resolved